spec/consumer_spec.rb

```diff
@@ -524,4 +526,35 @@
 
     it { expect(method_original_name).to eq(:trigger_heartbeat!) }
   end
+
+  describe '#interceptor' do
+    it "creates a consumer with interceptor" do
+      interceptor = FakeConsumerInterceptor.new
+      consumer = Kafka::Consumer.new(
+        cluster: cluster,
+        logger: logger,
+        instrumenter: instrumenter,
+        group: group,
+        offset_manager: offset_manager,
+        fetcher: fetcher,
+        session_timeout: session_timeout,
+        heartbeat: heartbeat,
+        interceptors: [interceptor]
+      )
+      expect(interceptor.init_count).to eq 1
```

Comment:
I don't see how `init_count` could even be any other number?

Reply:
if the interceptors weren't initialized
the java lib takes just the class name and initializes it internally. we ask the client to give us the interceptor already initialized 
maybe we should do this:
https://github.com/Shopify/ruby-kafka/pull/21#discussion_r437521678